src/test/java/com/fauna/e2e/E2EStreamingTest.java

```diff
@@ -106,4 +124,70 @@ public void query_streamOfPerson() throws InterruptedException {
         Integer total = products.stream().map(Product::getQuantity).reduce(0, Integer::sum);
         assertEquals(total, inventory.countInventory());
     }
+
+    @Test
+    public void handleLargeEvents() throws InterruptedException {
+        FaunaStream stream = client.stream(fql("Product.all().toStream()"), Product.class);
+        InventorySubscriber inventory = new InventorySubscriber();
+        stream.subscribe(inventory);
+        List<Product> products = new ArrayList<>();
+
+        byte[] image = new byte[20];
+        random.nextBytes(image);
+        // Product cheese = new Product("cheese", 1, image);
+        StringBuilder fifteenKName = new StringBuilder();
+        for (int i = 0; i < 1024 * 15; i++) {
+            fifteenKName.append(candidateChars.charAt(random.nextInt(candidateChars.length())));
+        }
+        assertEquals(fifteenKName.length(), 15360); // 15k string works.
+        products.add(client.query(fql("Product.create({name: ${name}, quantity: 1})",
+                Map.of("name", fifteenKName.toString())), Product.class).getData());
+
+        StringBuilder sixteenKName = new StringBuilder();
+        for (int i = 0; i < 1024 * 16; i++) {
+            sixteenKName.append(candidateChars.charAt(random.nextInt(candidateChars.length())));
+        }
+        assertEquals(sixteenKName.length(), 16384);
+
+        // 16k string causes the stream to throw.
```

Comment:
This is really important to identify but i'm not sure we have to solve it for GA because I don't think most customers will be streaming 16k strings. We def do not need to handle this for beta IMO

Reply:
It's a more fundamental property of the transport layer, I think: there are no guarantees that a event will arrive as a single network package. We do emit a `\n` character between events, which is how the JS driver accumulates and split events received: https://github.com/fauna/fauna-js/blob/main/src/http-client/fetch-client.ts#L146-L177

Reply:
Oh interesting, is there a spec or design doc that specifies the `\n` character.